embulk-core/src/main/java/org/embulk/plugin/PluginClassLoader.java

```diff
@@ -518,400 +327,8 @@ public InputStream getInputStream() throws IOException {
-                    } else {  // JAR entry size unavailable
-                        final ByteArrayOutputStream bytesStream = new ByteArrayOutputStream();
-                        final byte[] buffer = new byte[1024];
-                        long accumulatedSize = 0;
-                        while (true) {
-                            final long readSize;
-                            try {
-                                readSize = embeddedJarInputStream.read(buffer, 0, 1024);
-                            } catch (IOException ex) {
-                                // TODO: Notify this to reporters as far as possible.
-                                lastException = ex;
-                                System.err.println("Failed to load entry in embedded JAR: " + classResourceName);
-                                ex.printStackTrace();
-                                break jarEntries;  // Breaking from loading since this JAR looks broken.
-                            }
-
-                            if (readSize < 0) {
-                                break;
-                            }
-
-                            bytesStream.write(buffer, 0, (int) readSize);
-                            accumulatedSize += readSize;
-                        }
-                        actualSize = accumulatedSize;
-                        classResourceBytes = bytesStream.toByteArray();
-                    }
-
-                    final CodeSource codeSource = new CodeSource(codeSourceUrl, (CodeSigner[]) null);
-                    final Class<?> definedClass;
-                    try {
-                        definedClass = defineClass(className, classResourceBytes, 0, (int) actualSize, codeSource);
-                    } catch (Throwable ex) {
-                        // TODO: Notify this to reporters as far as possible.
-                        lastException = ex;
-                        System.err.println("Failed to load entry in embedded JAR: " + classResourceName);
-                        ex.printStackTrace();
-                        continue;
-                    }
-                    return definedClass;
-                }
-                try {
-                    jarEntry = embeddedJarInputStream.getNextJarEntry();
-                } catch (IOException ex) {
-                    // TODO: Notify this to reporters as far as possible.
-                    lastException = ex;
-                    System.err.println("Failed to load entry in embedded JAR: " + classResourceName);
-                    ex.printStackTrace();
-                    break jarEntries;  // Breaking from loading since this JAR looks broken.
-                }
-            }
-        }
-        if (lastException != null) {
-            throw new ClassNotFoundException(className, lastException);
-        } else {
-            throw new ClassNotFoundException(className);
-        }
-    }
-
     private InputStream getResourceAsStreamFromChild(final String resourceName) {
```

Comment:
We may be able to remove this method to change the caller side to call `super.getResourceAsStream(resourceName)` directly.

Also, the caller side `getResourceAsStream` may be removed. `URLClassLoader#getResourceAsStream` is actually calling `getResource` to find the resource as below.
http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/net/URLClassLoader.java#l232

`getResource` is overridden in `PluginClassLoader` in the same way where the overridden `getResourceAsStream` is doing.

Reply:
I just removed the method and call `super.getResourceAsStream(resourceName)` directly in ca688e9

Reply:
Thanks!

Plus, as a straight inference looking in `URLClassLoader.java`, I guessed we can remove the overridden `PluginClassLoader#getResourceAsStream` as well.

What the overridden `PluginClassLoader#getResourceAsStream` does is almost the same as the overridden `PluginClassLoader#getResource`, and then just loading it as `InputStream` in addition.

The same operation could be done just by the non-overridden `URLClassLoader#getResourceAsStream` calling `getResource`, where `getResource` is actually overridden as `PluginClassLoader#getResource`.

Reply:
I see. Removed it in 90ab1f6